main.go

```diff
@@ -6,492 +6,75 @@
-		if err != nil {
-			return err
-		}
-	}
-
-	diffs := rg.ChangedSince(prg)
-
-	of, err := os.Create(*fReportPrefix + "issues.html")
-	if err != nil {
-		return err
-	}
-	log.Print("Creating ", of.Name(), " (this may take a while)...")
-	if err := rg.ReportIssues(of); err != nil {
-		return err
-	}
-	of.Close()
-	filter, err := createFilterFromCmdLine()
-	if err != nil {
-		return err
-	}
-	if !filter.IsEmpty() || diffs != nil {
-		of, err := os.Create(*fReportPrefix + "issues-filtered.html")
-		if err != nil {
-			return err
-		}
-		log.Print("Creating ", of.Name(), " (this may take a while)...")
-		if err := rg.ReportIssuesFiltered(of, &filter, diffs); err != nil {
-			return err
-		}
-		of.Close()
-	}
-
-	return nil
-}
-
-// reportUp creates a requirements graph (and if necessary for comparison a previous graph) and
-// generates a bottom-up html report, showing the top-level requirement for each implemented function
-// @llr REQ-TRAQ-SWL-35
-func reportUp(reqtraqConfig *config.Config) error {
-	rg, err := buildGraph(*fAt, reqtraqConfig)
-	if err != nil {
-		return err
-	}
-
-	var prg *ReqGraph
-	if *fSince != "" {
-		prg, err = buildGraph(*fSince, reqtraqConfig)
-		if err != nil {
-			return err
-		}
-	}
-
-	diffs := rg.ChangedSince(prg)
-
-	of, err := os.Create(*fReportPrefix + "up.html")
-	if err != nil {
-		return err
-	}
-	log.Print("Creating ", of.Name(), " (this may take a while)...")
-	if err = rg.ReportUp(of); err != nil {
-		return err
-	}
-	of.Close()
-
-	filter, err := createFilterFromCmdLine()
-	if err != nil {
-		return err
-	}
-	if !filter.IsEmpty() || diffs != nil {
-		of, err := os.Create(*fReportPrefix + "up-filtered.html")
-		if err != nil {
-			return err
-		}
-		log.Print("Creating ", of.Name(), " (this may take a while)...")
-		if err := rg.ReportUpFiltered(of, &filter, diffs); err != nil {
-			return err
-		}
-		of.Close()
-	}
-
-	return nil
-}
-
-// showHelp prints usage information, either for the root command or for a sub-command
-// @llr REQ-TRAQ-SWL-32
-func showHelp() {
-	subCommand := ""
-	if len(os.Args) > 1 {
-		subCommand = os.Args[1]
-	}
-	switch subCommand {
-	case "help", "": // general help
-		fmt.Println(usage)
-	case "list":
-		fmt.Println(listUsage)
-	case "nextid":
-		fmt.Println(nextidUsage)
-	case "reportup", "reportdown", "reportissues":
-		fmt.Println(reportUsage)
-	case "validate":
-		fmt.Println(validateUsage)
-	case "web":
-		fmt.Println(webUsage)
-	default:
-		fmt.Printf("Unknown command '%s'", subCommand)
-		fmt.Println(usage)
-	}
-}
-
-// validate builds the requirement graph, gathering any errors and prints them out. If the strict flag is set return an error.
-// @llr REQ-TRAQ-SWL-36
-func validate(reqtraqConfig *config.Config) error {
-	rg, err := buildGraph(*fAt, reqtraqConfig)
-	if err != nil {
-		return err
-	}
-
-	if len(rg.Errors) > 0 {
-		for _, e := range rg.Errors {
-			fmt.Println(e)
-		}
-		if *fStrict {
-			return errors.New("ERROR. Validation failed")
+// @llr REQ-TRAQ-SWL-57
```

Comment:
What do you mean?, it is just for shell completions. Note that cobra includes a hidden `__complete` command that would invoke this (because this method is passed to `ValidArgsFunction` in the nextid and list commands).

Reply:
ahh sorry, my bad. disregard.